### PR TITLE
Movie details [IN PROGRESS]

### PR DESCRIPTION
**Pull request by gabichuelas**
@rrabinovitch 
This is just so you can check my code out while I continue to work on it. 
If at any point you want what I've done here, go ahead and merge and let me know. I will only push up to this PR work that is safe to merge, this way we don't have to deal with conflicts / doing more work than necessary as you work with related issues.

### CODE REVIEW

File: app/poros/movie.rb [hunk 1]
```diff
@@ -1,9 +1,22 @@
 class Movie
-  attr_reader :id, :title, :vote_avg, :overview
+  attr_reader :id, :title
   def initialize(movie_data)
     @id = movie_data[:id]
     @title = movie_data[:title]
-    @vote_avg = movie_data[:vote_average]
-    @overview = movie_data[:overview]
+    @raw_hash ||= MovieDBService.new.find_by(@id)
```
Comment (gabichuelas) [inline app/poros/movie.rb]: @rrabinovitch trying to think through this Movie model and how it works... when we create movie objects right now, they're coming from two different endpoints (and possibly more later on), so im wondering what is redundant and what isn't. In this case, the original `movie_data` hash from a particular endpoing may not have all the info we need, so I figured we could use the `MovieDBService` to pull the rest of the info? 

At that point, the decision is whether to also put it in `initialize`, or make separate methods. I'm not sure what the best option is in order to reduce API calls.

**Comment by gabichuelas** [comment]
@rrabinovitch 
Everything that needs to show on the Movie Details Page is there. Added a bunch of new methods to the DB Service, SearchResults, and Movie PORO to support the extra info, as well as creating two new POROs: `CastMember` and `Review`. 

Everything is there except a stubbed test for the Detail Page itself, which I will circle back to tomorrow. Feel free to merge so you can have all this code too!

PS. During Q&A tomorrow I'm hoping to spend some time talking to Ian about whether the way I'm getting the API info and percolating it through our classes is redundant. Tests seem to take a lot longer now (even though I didn't add any new ones) and I think it might be because of the way information is being called and displayed on the Movie Detail Page. Just something to think about..

File: app/controllers/movies_controller.rb [hunk 2]
```diff
@@ -2,9 +2,13 @@ class MoviesController < ApplicationController
   before_action :require_login
 
   def index
+    # if query params include keywords, perform that search,
+    # else, show top40 results (those are the two routes to
+    # get to this index action)
```
Comment (rrabinovitch) [inline app/controllers/movies_controller.rb:7]: perfect, this is the approach I'm going for

File: app/views/movies/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  <li><%= genre %></li>
+<% end %>
+</ul>
+<p>Cast (top 10):</p>
+<ul>
+<% @movie.first_ten_cast_members.each do |member| %>
+  <li><%= member.name %> -- as <%= member.character %></li>
+<% end %>
+</ul>
+<h4>Reviews(<%= @movie.reviews.count %>)</h4>
```
Comment (rrabinovitch) [inline app/views/movies/show.html.erb:19]: `count` might be a _slightly_ better fit as a PORO method so we're limiting the logic that happens in the view, and we may want to change it to `size` or `length` for optimization

File: spec/features/movies/show_spec.rb [hunk 4]
```diff
@@ -3,26 +3,41 @@
...
     allow_any_instance_of(ApplicationController).to receive(:current_user).and_return(user)
+
+    # json_response1 = File.read('spec/fixtures/top_rated_movie1.json')
+    # json_response2 = File.read('spec/fixtures/top_rated_movie2.json')
+    #
+    # stub_request(:get, "https://api.themoviedb.org/3/movie/top_rated?api_key=033cf2fe940e095f326c2f128d70fdfa&language=en-US&page=1").
+    #   to_return(status: 200, body: json_response1)
+    #
+    # stub_request(:get, "https://api.themoviedb.org/3/movie/top_rated?api_key=033cf2fe940e095f326c2f128d70fdfa&language=en-US&page=2").
+    #   to_return(status: 200, body: json_response2)
```
Comment (rrabinovitch) [inline spec/features/movies/show_spec.rb:14]: hopefully we can figure this out today!

**Review by rrabinovitch — APPROVED** [review]
This story seems like it was a big lift in figuring out the logic involved, awesome job!

**Pull request merged**